### PR TITLE
Add ip_collection to google_compute_forwarding_rule

### PR DESCRIPTION
**Pull request by zachberger**
Fixes  https://github.com/hashicorp/terraform-provider-google/issues/18407

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
COMPUTE: added `ip_collection` field to `google_compute_forwarding_rule` resource
```

### CODE REVIEW

File: mmv1/templates/terraform/examples/go/forwarding_rule_externallb_ipv6.tf.tmpl [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  name                  = "{{index $.Vars "forwarding_rule_name"}}"
+  region                = "us-west1"
+  port_range            = 80
+  ip_protocol           = "TCP"
+  ip_version            = "IPV6"
+  load_balancing_scheme = "EXTERNAL"
+  ip_address            = "2600:1901:4456::1/96"
+  network_tier          = "PREMIUM"
+  backend_service       = google_compute_region_backend_service.backend.id
+  ip_collection         = "projects/zberger-sandbox-276421/regions/us-west1/publicDelegatedPrefixes/tf-test-prefix"
```
Comment (zachberger) [inline mmv1/templates/terraform/examples/go/forwarding_rule_externallb_ipv6.tf.tmpl:13]: Needs to be updated to use a PDP in the CI project

File: mmv1/templates/terraform/examples/forwarding_rule_externallb_ipv6.tf.erb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  name                  = "<%= ctx[:vars]['forwarding_rule_name'] %>"
+  region                = "us-west1"
+  port_range            = 80
+  ip_protocol           = "TCP"
+  ip_version            = "IPV6"
+  load_balancing_scheme = "EXTERNAL"
+  ip_address            = "2600:1901:4456::1/96"
+  network_tier          = "PREMIUM"
+  backend_service       = google_compute_region_backend_service.backend.id
+  ip_collection         = "projects/zberger-sandbox-276421/regions/us-west1/publicDelegatedPrefixes/tf-test-prefix"
```
Comment (zachberger) [inline mmv1/templates/terraform/examples/forwarding_rule_externallb_ipv6.tf.erb:12]: Needs to be updated to use a PDP in the CI project

File: mmv1/products/compute/PublicAdvertisedPrefix.yaml [hunk 3]
```diff
@@ -51,6 +51,16 @@ examples:
       prefixes_name: 'my-prefix'
     test_env_vars:
       desc: :PAP_DESCRIPTION
+  - !ruby/object:Provider::Terraform::Examples
```
Comment (melinath) [inline mmv1/products/compute/PublicAdvertisedPrefix.yaml:54]: could you make the changes to the prefix resources in a separate PR please?

Comment (zachberger) [reply mmv1/products/compute/PublicAdvertisedPrefix.yaml:54]: I've opened #11406. Will pull the changes out of here shortly.

**Comment by rileykarson** [comment]
Hey! I'm closing this PR as a part of a cleanup of older inactive PRs, using a threshold of PRs in "Draft" last updated over a month ago. This doesn't represent rejection of the change, and feel free to comment for me to reopen it if you plan to pick it back up, or feel free to start a new PR with the same changes in the future.

**Pull request closed without merging**